File: presentation/src/main/java/com/popularmovies/vpaliy/popularmoviesapp/ui/media/MoviesFragment.java
package com.popularmovies.vpaliy.popularmoviesapp.ui.media;


import com.popularmovies.vpaliy.domain.configuration.SortType;
import com.popularmovies.vpaliy.popularmoviesapp.App;
import com.popularmovies.vpaliy.popularmoviesapp.R;
import com.popularmovies.vpaliy.popularmoviesapp.di.component.DaggerViewComponent;
import com.popularmovies.vpaliy.popularmoviesapp.di.module.PresenterModule;
import com.popularmovies.vpaliy.popularmoviesapp.ui.media.MediaContract.Presenter;
import java.util.Arrays;
import java.util.List;

import android.graphics.Color;
import android.support.annotation.NonNull;
import com.popularmovies.vpaliy.data.source.qualifier.Movies;
import com.popularmovies.vpaliy.popularmoviesapp.ui.base.MediaType;

import javax.inject.Inject;

public class MoviesFragment extends MediaFragment {

    @Override @Inject @Movies
    public void attachPresenter(@NonNull @Movies Presenter presenter) {
        this.presenter=presenter;
        this.presenter.attachView(this);
    }

    @Override
    public void initializeDependencies() {
        DaggerViewComponent.builder()
                .applicationComponent(App.appInstance().appComponent())
                .presenterModule(new PresenterModule())
                .build().inject(this);

    }

    @Override
    public int getColor(SortType sortType) {
        return Color.parseColor("#ff6f00");
    }

    @Override
    public List<SortType> getSortTypes() {
        return Arrays.asList(SortType.POPULAR, SortType.LATEST,
                SortType.NOW_PLAYING, SortType.UPCOMING,SortType.TOP_RATED);
    }

    @Override
    public MediaType getMediaType() {
        return MediaType.MOVIES;
    }

    @Override
    public String getTitle(SortType sortType) {
        switch (sortType){
            case POPULAR:
                return getString(R.string.popular_media);
            case LATEST:
                return getString(R.string.latest_media);
            case NOW_PLAYING:
                return getString(R.string.now_playing_media);
            case UPCOMING:
                return getString(R.string.upcoming_media);
            case TOP_RATED:
                return getString(R.string.top_rated_media);
            default:
                throw new IllegalArgumentException();
        }
    }
}

File: presentation/src/main/java/com/popularmovies/vpaliy/popularmoviesapp/ui/home/MoviesFragment.kt
package com.popularmovies.vpaliy.popularmoviesapp.ui.home

import com.popularmovies.vpaliy.domain.entity.MediaType
import com.popularmovies.vpaliy.popularmoviesapp.App
import com.popularmovies.vpaliy.popularmoviesapp.R
import com.popularmovies.vpaliy.popularmoviesapp.di.component.DaggerMovieComponent
import com.popularmovies.vpaliy.popularmoviesapp.di.module.MovieModule
import com.vpaliy.kotlin_extensions.getColor
import javax.inject.Inject

class MoviesFragment : HomeFragment() {
  override var presenter: HomeContract.Presenter? = null
    @Inject set(value) {
      field = value
      field?.attach(this)
    }

  private val typeMap by lazy(LazyThreadSafetyMode.NONE) {
    HashMap<MediaType, String>().apply {
      put(MediaType.POPULAR, getString(R.string.popular_media))
      put(MediaType.TOP, getString(R.string.top_rated_media))
      put(MediaType.UPCOMING, getString(R.string.upcoming_media))
      put(MediaType.NOW_PLAYING, getString(R.string.now_playing_media))
    }
  }

  override fun types() = typeMap.keys.toTypedArray()

  override fun getColor(type: MediaType) = getColor(R.color.colorMovies)

  override fun getTitle(type: MediaType) = typeMap[type]!!

  override fun inject() {
    DaggerMovieComponent.builder()
        .applicationComponent(App.component)
        .movieModule(MovieModule())
        .build().inject(this)
  }
}